File: app/src/main/java/com/revlogix/app/data/local/Vehicle.kt
package com.revlogix.app.data.local

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(
    tableName = "vehicles",
    foreignKeys = [
        ForeignKey(
            entity = User::class,
            parentColumns = ["userId"],
            childColumns = ["userId"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class Vehicle(
    @PrimaryKey(autoGenerate = true)
    val vehicleId: Int = 0,
    val userId: Int,
    val make: String,
    val model: String,
    val year: Int,
    val bodyStyle: String,
    val registrationNumber: String,
    val currentOdometer: Int,
    val dateAdded: Long = System.currentTimeMillis()
)

File: RevLogixApi/Models/Vehicle.cs
namespace RevLogixApi.Models;

public class Vehicle
{
    public int VehicleId { get; set; }
    public int UserId { get; set; }
    public string Make { get; set; } = "";
    public string Model { get; set; } = "";
    public int Year { get; set; }
    public string BodyStyle { get; set; } = "";
    public string RegistrationNumber { get; set; } = "";
    public int CurrentOdometer { get; set; }
}